File: 2006-find-the-student-that-will-replace-the-chalk/solution.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        summ=sum(chalk)
        n=len(chalk)
        #cycle=summ//n
        #newk=k-cycle*summ
        k%=summ
        ans=0
        for i in range(n):
            if(k < chalk[i]):
                ans=i
                break
            k-=chalk[i]
        return ans
                

        

File: 2006-find-the-student-that-will-replace-the-chalk/solution.cpp
class Solution {
public:
    int chalkReplacer(vector<int>& chalk, int k) {
        long long n=chalk.size();
        long long summ=0;
        for(int i=0;i<n;i++){
            summ+=chalk[i];
        }
       /* long long cycle=floor(summ/n);
        long long newk=k-cycle*summ;*/
        k%=summ;
        long long ans=0;
        for(int i=0;i<n;i++){
            if(k < chalk[i]){
                ans=i;
                break;
            }
            k-=chalk[i];
        }
        return ans;
    }
};
